src/extensions/LinkSiteTree.php

```diff
@@ -131,16 +157,33 @@ public function updateIsOrphaned(&$status)
             isset($owner->SiteTreeID) &&
             $owner->CurrentPage instanceof SiteTree &&
             $currentPage = $owner->CurrentPage
-        ){
+        ) {
             // Always false for root pages
             if (empty($owner->SiteTree()->ParentID)) {
                 $status = false;
                 return;
             }
 
             // Parent must exist and not be an orphan itself
-            $parent = $this->Parent();
+            $parent = $owner->Parent();
             $status = !$parent || !$parent->exists() || $parent->isOrphaned();
         }
     }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    public function onBeforeWrite()
+    {
+        parent::onBeforeWrite();
+
+        $owner = $this->getOwner();
+
+        if (empty($owner->Title) && $owner->Type === "SiteTree" && $owner->SiteTreeID && $owner->ElementID && $element = $owner->Element()) {
+            $owner->Title = $element->Title;
+        }
+        $owner->Querystring = $owner->Querystring ? str_replace('?', '', (string) $owner->QueryString) : $owner->Querystring;
+        $owner->Anchor = $owner->Anchor ? str_replace('#', '', (string) $owner->Anchor) : $owner->Anchor;
```

Comment:
This looks overly complicated.
A simple `$owner->Querystring = ltrim((string)$owner->Querystring, '?');` would suffice.